File: src/au/com/rmit/Game2dEngine/action/Action.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package au.com.rmit.Game2dEngine.action;

import au.com.rmit.Game2dEngine.node.MovingSprite;
import java.util.Random;

/**
 *
 * @author ricolwang
 */
public abstract class Action
{

    public String identifer = this.getClass().getName();
    protected boolean bImmediately;

    public static final double MINIMUM = 0.001;
    public static final double EQUAL_STANDARD = 0.000000001;
    public static final double MINIMUM_DURATION = 0.001;

    public Random theRandom = new Random();
    protected MovingSprite theSprite = null;
    public boolean bComplete = false;

    //runningTime in milliseconds
    public void perform(double runningTime)
    {
        bComplete = theRandom.nextBoolean();
    }

    public void setSprite(MovingSprite theSprite)
    {
        this.theSprite = theSprite;
    }

    public void clearSprite()
    {
        this.theSprite = null;
    }

    @Override
    public String toString()
    {
        return "Action: " + this.getClass().getName() + " - id: " + identifer;
    }
}

File: src/com/wang/Game2dEngine/action/Action.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.wang.Game2dEngine.action

import com.wang.Game2dEngine.sprite.Sprite
import java.util.*

/**
 * @author ricolwang
 */
abstract class Action
{
    var identifier = this.javaClass.name
    protected var bImmediately = false
    var theRandom = Random()
    var theSprite: Sprite? = null
        protected set
    var bComplete = false

    //runningTime in milliseconds
    open fun perform(runningTime: Double)
    {
        bComplete = theRandom.nextBoolean()
    }

    fun setSprite(aSprite: Sprite?)
    {
        theSprite = aSprite
    }

    open fun clearSprite()
    {
        theSprite = null
    }

    override fun toString(): String
    {
        return "Action: " + this.javaClass.name + " - id: " + identifier
    }

    companion object
    {
        const val MINIMUM = 1e-03
        const val EQUAL_STANDARD = 1e-10
        const val MINIMUM_DURATION = 1e-3
    }
}